File: 1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.java
import java.util.*;

class Solution {
    public boolean uniqueOccurrences(int[] arr) {
        boolean[] alreadyExistNumberOfOccurrences = new boolean[arr.length + 1];
        Map<Integer, Integer> countNumberOfOccurrences = new HashMap<>();  

        for (int i = 0; i < arr.length; i++) {
            int numberOfOccurrences = countNumberOfOccurrences.getOrDefault(arr[i], 0);
            countNumberOfOccurrences.put(arr[i], numberOfOccurrences + 1);
        }

        for (Integer key : countNumberOfOccurrences.keySet()) {
            if (alreadyExistNumberOfOccurrences[countNumberOfOccurrences.get(key)]) {
                return false;
            }

            alreadyExistNumberOfOccurrences[countNumberOfOccurrences.get(key)] = true;
        }

        return true;
    }
}

File: 1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.ts
function uniqueOccurrences(arr: number[]): boolean {
  let m = new Map<number, number>();
  let s = new Set<number>();

  for (let i = 0; i < arr.length; i++) {
    m.set(arr[i], (!m.get(arr[i]) ? 0 : m.get(arr[i])) + 1);
  }

  m.forEach((value) => s.add(value));
  return m.size === s.size;
}
